api/api.go

```diff
@@ -0,0 +1,64 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/gorilla/context"
+	"github.com/mailgun/oxy/forward"
+)
+
+type (
+	// API services a set of HTTP routes
+	API struct {
+		listenAddr string
+		uiDir      string
+		dockerURL  string
+		fwd        *forward.Forwarder
+	}
+
+	// Config for constructing an API
+	Config struct {
+		ListenAddr string
+		UIDir      string
+		DockerURL  string
+	}
+)
+
+// NewAPI configures and creates an API instance
+func NewAPI(config Config) (*API, error) {
```

Comment:
From the above comment, this can become a pointer and then just assign it to the single `config` field in the API struct.